websites/C/CodinGame/presence.ts

```diff
@@ -0,0 +1,199 @@
+				}/${document.querySelector("span[class='rank-total']").textContent})`;
+			}
+		} else presenceData.details = "Looking at Compete";
+	} else if (path.includes("/cooperate"))
+		presenceData.details = "Looking at Cooperate";
+	else if (path.includes("/learn")) presenceData.details = "Looking at Learn";
+	else if (path.includes("/events")) presenceData.details = "Looking at Events";
+	else if (path.includes("/contribute/community"))
+		presenceData.details = "Looking at Community";
+	else if (path.includes("/contribute/my-contributions"))
+		presenceData.details = "Looking at My Contributions";
+	else if (path.includes("/live-streams"))
+		presenceData.details = "Looking at Live Streams";
+	else if (path.includes("/blog")) presenceData.details = "Looking at Blog";
+	else if (path.includes("/forum")) presenceData.details = "Looking at Forum";
+	else if (path.includes("/profile"))
+		presenceData.details = "Looking at Profile";
+	else if (path.includes("/settings"))
+		presenceData.details = "Looking at Settings";
+	else if (path.includes("/about")) presenceData.details = "Looking at Abouts";
+	else if (path.includes("/careers"))
+		presenceData.details = "Looking at Careers";
+	else if (path.includes("/work")) presenceData.details = "Looking at Work";
+	else if (
+		path.includes("/legal/privacy-policy") &&
+		path.includes("/playgrounds")
+	)
+		presenceData.details = "Looking at Privacy Policy";
+	else if (path.includes("/help-center"))
+		presenceData.details = "Looking at FAQ";
+	else if (path.includes("/start")) presenceData.details = "Not logged in";
+	else if (path.includes("/ide")) {
+		if (path.includes("/ide/puzzle/")) {
+			presenceData.details = `Solving the challenge : ${
+				document.querySelectorAll("h1[class='cg-ide-title'] > span")[
+					document.querySelectorAll("h1[class='cg-ide-title'] > span").length -
+						1
+				].textContent
+			}`;
+			try {
+				const score = document.querySelector(
+					"span[class*='score-value']"
+				).textContent;
+				if (score === "N/A") presenceData.state = "(0%)";
+				else {
+					try {
+						const rank = document.querySelector(
+							"span[class='rank-value']"
+						).textContent;
+						if (rank !== "") {
+							presenceData.state = `${rank}th/${
+								document.querySelector("span[class='rank-total']").textContent
+							}`;
+						} else presenceData.state = `(${score}%)`;
+					} catch {
+						presenceData.state = `(${score}%)`;
+					}
+				}
+			} catch {
+				presenceData.state = `League : ${
+					document.querySelector("span[class='league-value']").textContent
+				}`;
+			}
+		} else {
+			presenceData.details = document.querySelectorAll(
+				"h1[class='cg-ide-title'] > span"
+			)[
+				document.querySelectorAll("h1[class='cg-ide-title'] > span").length - 1
+			].textContent;
+			presenceData.state = `Time left : ${
+				document.querySelector("span[class*='minutes']").textContent
+			}m : ${document.querySelector("span[class*='seconds']").textContent}s`;
+		}
```

Comment:
You don't have to do this, but you could use timestamps if successfully converted

Reply:
Yes, but with this I can get exactly the time before the clash is over.